release notes/v0.47.0.md

```diff
@@ -0,0 +1,92 @@
+
+### UX improvements and enhancements
+
+- [#3338](https://github.com/grafana/k6/pull/3338), [xk6-grpc#48](https://github.com/grafana/xk6-grpc/pull/48). Adds a support of the v1 of the gRPC reflection protocol.
+- [#3290](https://github.com/grafana/k6/pull/3290) Log errors when executing `setup` and `teardown` via REST API.
+- [#3327](https://github.com/grafana/k6/pull/3327) `k6 version` now prints the commit identifier for the build.
+- [#3340](https://github.com/grafana/k6/pull/3340) The k6 ``*-with-browser`` Docker images set now the `no-sandbox` environment variable automatically.
+
+## Bug fixes
+
+- [xk6-grpc#47](https://github.com/grafana/xk6-grpc/pull/47) Fix the premature closing of the stream when a stream's client has finished sending. Thanks, @thiagodpf, for reporting!
+- [#3344](https://github.com/grafana/k6/pull/3344), [xk6-grpc#49](https://github.com/grafana/xk6-grpc/pull/49) adds (fixes) the support of google protobuf wrappers. Thanks, @zibul444, for reporting!
+- [#3346](https://github.com/grafana/k6/pull/3346) A data race when `http.batch` it is used with binary response has been fixed.
+- [#3308](https://github.com/grafana/k6/pull/3308) `goja` version has been updated and it fixes a compiler bug when class is declared in a function with argument.
+- [browser#1040](https://github.com/grafana/xk6-browser/pull/1040) fix the `context.clearCookies()` method to clear all cookies from the current browser context.
+- [browser#1031](https://github.com/grafana/xk6-browser/pull/1031) fix the `expires` field while adding cookies using the `context.addCookie()` method.
+- [browser#1039](https://github.com/grafana/xk6-browser/pull/1039) fix Goja conversions while adding and retrieving cookies.
+
+## Maintenance and internal improvements
+
+- [#3342](https://github.com/grafana/k6/pull/3342) Updates xk6-grpc to the latest version. This change brings all the latest fixes and improvements to the experimental gRPC module.
+- [#3271](https://github.com/grafana/k6/pull/3271),[#3272](https://github.com/grafana/k6/pull/3272) Updated the golangci version and added the `interfacebloat` linter.
+- [#3279](https://github.com/grafana/k6/pull/3279) Fixed the CI not publishing the SBOM file on new a release.
+- [#3283](https://github.com/grafana/k6/pull/3283) Updated the Go version to 1.21.
+- [#3351](https://github.com/grafana/k6/pull/3351) Updated the version of Prometheus remote write output. Check the specific [release notes](https://github.com/grafana/xk6-output-prometheus-remote/releases/tag/v0.3.0).
+- [#3341](https://github.com/grafana/k6/pull/3341), [#3339](https://github.com/grafana/k6/pull/3339) Updated `goja` that includes Runtime initialization speed-up and a fix for source indexes. 
+- [#3311](https://github.com/grafana/k6/pull/3311) Updated the `alpine` image version that is used as the base of the k6 Docker image.
+
+## _Optional_ Roadmap
+
+_Discussion of future plans_
```

Comment:
@mstoykov do you want to include the ECMAScript Modules here?

Reply:
@mstoykov can you make a suggestion for it, please? You are probably the best for doing it.

Reply:
```suggestion
### Native Modules Support

Work on this [epic issue](https://github.com/grafana/k6/issues/3265) has been picked up and there is some progress in the underlying implementation. 

One of the main internal changes will be dropping of our internal [babel](https://babeljs.io/) that is currently used to transpile [ESM](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) code to [commonjs]()https://en.wikipedia.org/wiki/CommonJS) one.
 
For users that will mean better JS support as this for example will automatically get [object spread](https://github.com/grafana/k6/issues/824) working and likely faster startup for *big* scripts. In the future this also means easier adding of JS compatibility as now it needs to be supported only in the JS VM we use - [goja](https://github.com/dop251/goja)

There is risk though that some k6 tests using both commonjs and ESM syntax will change in behavior. This in practice should never really be done as both are not compatible.

Due to this we are going to be dropping it very carefully and likely only once we have done a lot of tests. This currently means that it is planned for v0.49.0, but with the majority of the remaining work done in v0.48.0 cycle.

This way as well we will have the most time of users being able to test from our `master` branch, using docker images for example.
```
I tried to give some amount of info so it is less "we are doing this - figure it out", but also tried not to list everything.